app/helpers/fhir_helper.rb

```diff
@@ -0,0 +1,329 @@
+# frozen_string_literal: true
+
+# Helper module for FHIR translations
+module FhirHelper # rubocop:todo Metrics/ModuleLength
+  # Returns a representative FHIR::Patient for an instance of a Sara Alert Patient. Uses US Core
+  # extensions for sex, race, and ethnicity.
+  # https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-patient.html
+  def patient_as_fhir(patient)
+    creator_agent_ref = FHIR::Reference.new
+    creator_app = OauthApplication.where(user_id: patient.creator.id).first
+    if creator_app
+      # Created with an M2M workflow client
+      creator_agent_ref.identifier = FHIR::Identifier.new(value: creator_app.uid)
+      creator_agent_ref.display = creator_app.name
+    else
+      # Created with a user worfklow client
+      creator_agent_ref.identifier = FHIR::Identifier.new(value: patient.creator.id)
+      creator_agent_ref.display = patient.creator.email
+    end
+
+    FHIR::Patient.new(
+      meta: FHIR::Meta.new(lastUpdated: patient.updated_at.strftime('%FT%T%:z')),
+      contained: [FHIR::Provenance.new(
+        # Would like to use a Rails URL Helper here, but we don't get one for this endpoint
+        target: [FHIR::Reference.new(reference: "/fhir/r4/Patient/#{patient.id}")],
+        agent: [
+          FHIR::Provenance::Agent.new(
+            who: creator_agent_ref
+          )
+        ],
+        recorded: patient.created_at.strftime('%FT%T%:z'),
+        activity: {
+          coding: [
+            {
+              system: 'http://terminology.hl7.org/CodeSystem/v3-DataOperation',
+              code: 'CREATE',
+              display: 'create'
+            }
+          ]
+        }
+      )],
+      id: patient.id,
+      active: patient.monitoring,
+      name: [FHIR::HumanName.new(given: [patient.first_name, patient.middle_name].reject(&:blank?), family: patient.last_name)],
+      telecom: [
+        patient.primary_telephone ? FHIR::ContactPoint.new(system: 'phone',
+                                                           value: patient.primary_telephone,
+                                                           rank: 1,
+                                                           extension: [to_string_extension(patient.primary_telephone_type, 'phone-type')])
+                                  : nil,
+        patient.secondary_telephone ? FHIR::ContactPoint.new(system: 'phone',
+                                                             value: patient.secondary_telephone,
+                                                             rank: 2,
+                                                             extension: [to_string_extension(patient.secondary_telephone_type, 'phone-type')])
+                                    : nil,
+        patient.email ? FHIR::ContactPoint.new(system: 'email', value: patient.email, rank: 1) : nil
+      ].reject(&:nil?),
+      birthDate: patient.date_of_birth&.strftime('%F'),
+      address: [
+        FHIR::Address.new(
```

Comment:
How do we handle Patients that have a foreign address that they enter? Or both foreign and domestic addresses?
I would expect we'd need to at least check for the foreign address if the domestic address is blank.

Reply:
Sounds reasonable to me. If they have both we could include both. This wasn't actually something I added as part of this task though (it's just showing up in this diff because I moved it ([context](https://github.com/SaraAlert/SaraAlert/pull/521#issuecomment-728112422)), so I'll create a ticket for this change.

Reply:
https://tracker.codev.mitre.org/browse/SARAALERT-1032

Reply:
That's fair - thanks!